### PR TITLE
Add cancellation support

### PR DESCRIPTION
**Pull request by Tarrowren**
issue #10

### CODE REVIEW

**Comment by Tarrowren** [comment]
@microsoft-github-policy-service agree

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "http-proxy-agent": "^5.0.0",
     "https-proxy-agent": "^5.0.0",
     "vscode-nls": "^5.0.0",
+    "vscode-jsonrpc": "^5.0.0",
```
Comment (aeschli) [inline package.json:23]: I'd rather not add any new dependencies. It's ok to duplicate the types and implementations.

File: src/browser/main.ts [hunk 2]
```diff
@@ -8,47 +8,58 @@ import { XHRRequest, XHRConfigure, XHROptions, XHRResponse } from '../../api';
...
-	const requestInit: RequestInit = {
-		method: options.type,
-		redirect: options.followRedirects > 0 ? 'follow' : 'manual',
-		mode: 'cors',
-		headers: requestHeaders
-	};
-	if (options.data) {
-		requestInit.body = options.data;
-	}
+    const requestHeaders = new Headers();
```
Comment (aeschli) [inline src/browser/main.ts:11]: Please avoid any reformatting changes.Use tab.

**Review by aeschli — CHANGES_REQUESTED** [review]

**Comment by aeschli** [comment]
Thanks a lot for the PR, @Tarrowren! See my comments...

File: package.json (hunk 1 above)
Comment (Tarrowren) [reply package.json:23]: This is needed for unit testing, and this package is added to `devDependencies` and does not increase the size of the product

File: src/browser/main.ts (hunk 2 above)
Comment (Tarrowren) [reply src/browser/main.ts]: done

**Comment by aeschli** [comment]
The problem is that the api.d.ts has a `import("vscode-jsonrpc").CancellationToken;`.
As vscode-jsonrpc is only a dev devdependency that means that users of `request-light` wont get it, and the type can not be resolved. 
I see we have the same problem with HttpProxyAgent and HttpsProxyAgent.

I think we should redefine the intrefaces for CancellationToken (and Event and Disposable)

**Review by aeschli — APPROVED** [review]

**Review by aeschli — APPROVED** [review]

**Review by dbaeumer — APPROVED** [review]

**Pull request merged**